Eventy/src/main/java/org/example/eventy/events/repositories/EventRepository.java

```diff
@@ -49,4 +51,22 @@ List<Event> findOrganizedEventsByUserBetween(
     List<Event> findAttendingEventsByUserBetween(@Param("userId") Long userId,
                                                  @Param("startDate") LocalDate startDate,
                                                  @Param("endDate")LocalDate endDate);
+
+    @Query("SELECT e FROM Event e " +
+            "WHERE (:search = '' OR LOWER(e.name) = LOWER(:search)) " +
+            "   OR (:search = '' OR LOWER(e.description) = LOWER(:search)) " +
+            "AND (:maxParticipants IS NULL OR e.maxNumberParticipants <= :maxParticipants) " +
+            "AND (:location IS NULL OR e.location.name = :location) " +
+            "AND (CAST(:startDate AS timestamp) IS NULL OR CAST(:endDate AS timestamp) IS NULL OR e.date BETWEEN CAST(:startDate AS timestamp) AND CAST(:endDate AS timestamp)) " +
+            "AND (:eventTypes IS NULL OR e.type.name IN :eventTypes)")
+    Page<Event> findAll(@Param("search") String search,
+                        @Param("eventTypes") List<String> eventTypes,
+                        @Param("maxParticipants") Integer maxParticipants,
+                        @Param("location") String location,
+                        @Param("startDate") LocalDateTime startDate,
+                        @Param("endDate") LocalDateTime endDate,
+                        Pageable pageable);
+
+    @Query("SELECT e FROM Event e ORDER BY e.id DESC")
+    ArrayList<Event> findFeaturedEvents(Pageable pageable);
```

Comment:
Maybe add a limit here? So it limits to last 5 elements (so that only 5 elements are featured?